resources/tests/type_checker_tests/InState.obs

```diff
@@ -51,13 +51,13 @@ main contract InState {
              s.turnOff();
         }
 
-        // TODO: Make the type checker smart enough for this assertion to pass.
-        // [s @ Off];
+        [s @ Off];
 
         if (s in Off) {
              s.turnOn();
         }
 
+        // ERROR: s is On now, we know this specifically
```

Comment:
Need to update the unit tests to expect this error.

Reply:
Where can I find the unit tests?

Reply:
Look at TypeCheckerTests.scala. Or run sbt test and you'll see what fails.